pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -547,18 +554,32 @@ func (os *OpenStack) getDevicePathFromInstanceMetadata(volumeID string) string {
 }
 
 // GetDevicePath returns the path of an attached block storage volume, specified by its id.
-func (os *OpenStack) GetDevicePath(volumeID string) string {
-	devicePath := os.GetDevicePathBySerialID(volumeID)
-
-	if devicePath == "" {
+func (os *OpenStack) GetDevicePath(volumeID string) (string, error) {
+	backoff := wait.Backoff{
+		Duration: operationFinishInitDelay,
+		Factor:   operationFinishFactor,
+		Steps:    operationFinishSteps,
+	}
+
+	var devicePath string
+	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
```

Comment:
Try the operation a few times, sometimes when the status turns to `in-use` we still don't have the info we need